pkg/p2p/mock/mock.go

```diff
@@ -44,6 +45,12 @@ func WithPeersFunc(f func() []p2p.Peer) Option {
 	})
 }
 
+func WithSetPeerAddedHandlerFunc(f func(func(ctx context.Context, overlay swarm.Address) error)) Option {
```

Comment:
nice to have: extract the function signature to a single definition that you can reuse